backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/C_Location_StepDef.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * #%L
+ * de.metas.cucumber
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.cucumber.stepdefs;
+
+import de.metas.common.util.CoalesceUtil;
+import de.metas.util.Services;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.And;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.model.I_C_Location;
+
+import java.util.Map;
+
+import static de.metas.cucumber.stepdefs.StepDefConstants.TABLECOLUMN_IDENTIFIER;
+import static org.adempiere.model.InterfaceWrapperHelper.newInstanceOutOfTrx;
+import static org.assertj.core.api.Assertions.*;
+import static org.compiere.model.I_C_Location.COLUMNNAME_C_Location_ID;
+
+public class C_Location_StepDef
+{
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+	private final StepDefData<I_C_Location> locationTable;
+
+	public C_Location_StepDef(@NonNull final StepDefData<I_C_Location> locationTable)
+	{
+		this.locationTable = locationTable;
+	}
+
+	@And("metasfresh contains C_Location:")
+	public void add_C_Location(@NonNull final DataTable dataTable)
+	{
+		for (final Map<String, String> row : dataTable.asMaps())
+		{
+			final String address1 = DataTableUtil.extractStringForColumnName(row, I_C_Location.COLUMNNAME_Address1);
```

Comment:
- AFAIR the country is manadatory, while address1, postal and city are not. So i suggest that when adding this step, you 
  - directly add all the fields that the user can set in the UI
  - add the non-mandatory ones as optiona

Reply:
refactored as suggested